### DISCUSSION TITLE
Microsoft Dolby Digital Atmos Pack standalone installer - source code?

### DISCUSSION THREAD

**Discussion by rjappleton** [SimonMacer/AnWave#31] [Open Discussion]
Is the source code available for the standalone installer? I was hoping to add the functionality to my existing InnoSetup installer so that the AC-3 decoder could be silently installed if the user's PC was running 24H2 or later - the user wouldn't then have to be bothered with the problem at all.

The releases page has a "source code.zip" file but it just seems to contain a couple of .md files.

I suspect that just replacing the dolby DLL file in the Windows folder would not work - there are probably other registry tweaks needed, and the potential of screwing something up.

**Comment by SimonMacer** [comment]
> Is the source code available for the standalone installer? I was hoping to add the functionality to my existing InnoSetup installer so that the AC-3 decoder could be silently installed if the user's PC was running 24H2 or later - the user wouldn't then have to be bothered with the problem at all.
> 
> The releases page has a "source code.zip" file but it just seems to contain a couple of .md files.
> 
> I suspect that just replacing the dolby DLL file in the Windows folder would not work - there are probably other registry tweaks needed, and the potential of screwing something up.

If you want to make a InnoSetup installer, you need to put 64-bit/32-bit msac3enc.dll, DolbyDecMFT.dll and msmpeg2ac3dec.dll into %SYSTEM32/SYSWOW64%, and then register these Dlls through the regsvr32.exe /s command.

Then take TrustedInstaller permissions to import these *.reg files:
https://drive.google.com/file/d/1kJ9VdAPbAFHi5OSal3stQDWqCIHA6B0J/view?usp=sharing
`Microsoft Dolby Digital Encoder MFT Registry.reg`
`Microsoft Dolby Digital Plus Decoder MFT Registry.reg`
`Microsoft DSHOW AC3 Decoder Registry.reg`

and command:
REG ADD HKEY_LOCAL_MACHINE\SOFTWARE\Classes\CLSID\{083863F1-70DE-11D0-BD40-00A0C911CE86}\Instance\{5261169D-9B6C-435F-B1D5-F79BAF700C71}\ /v FriendlyName /t REG_SZ /d "Microsoft DSHOW AC3 Decoder" /f
REG ADD HKEY_LOCAL_MACHINE\SOFTWARE\Classes\WOW6432Node\CLSID\{083863F1-70DE-11D0-BD40-00A0C911CE86}\Instance\{5261169D-9B6C-435F-B1D5-F79BAF700C71}\ /v FriendlyName /t REG_SZ /d "Microsoft DSHOW AC3 Decoder" /f

**Reply by rjappleton** [reply]
Thanks Simon, I'll look into that.

I don't have a computer that has a clean install of 24H2, but someone who has, has sent me a DxDiag.exe output text file that I've compared with that on my PC that is on 23H2. Surprisingly, "Microsoft Dolby Digital Plus Decoder MFT" is still there on a clean install of 24H2, but its CLSID has changed. Below are the lines from DxDiag - the first line is from my PC and the second line is the replacement from the clean 24H2 PC:

Microsoft Dolby Digital Plus Decoder MFT, {177C0AFE-900B-48D4-9E4C-57ADD250B3D4}, 0x1, DolbyDecMFT.dll, 10.00.22621.4391
Microsoft Dolby Digital Plus Decoder MFT, {0C0EA61D-74FB-449B-8CA4-1DDF1BA9AA47}, 0x1, DolbyDecMFT.dll, 10.00.26100.2161

The name is exactly the same but the CLSID is different.

Surely Microsoft wouldn't have disabled the decoder just by changing its CLSID ?

**Reply by SimonMacer** [reply]
> {0C0EA61D-74FB-449B-8CA4-1DDF1BA9AA47}

The {0C0EA61D-74FB-449B-8CA4-1DDF1BA9AA47} is older CLSID. The CLSID used in 24H2 before Microsoft decided to remove AC-3.

Both CLISDs work with AC-3 MFT.

**Reply by rjappleton** [reply]
OK, so {0C0EA61D-74FB-449B-8CA4-1DDF1BA9AA47} won't work either?

**Reply by SimonMacer** [reply]
> OK, so {0C0EA61D-74FB-449B-8CA4-1DDF1BA9AA47} won't work either?

it also works. But you can use {177C0AFE-900B-48D4-9E4C-57ADD250B3D4}, only the MFT of 22621.XXXX CLSID is still active.

**Comment by rjappleton** [comment]
Pre-24H2 only sseems to have  {177C0AFE-900B-48D4-9E4C-57ADD250B3D4} listed in DxDiag, and 24H2 only has {0C0EA61D-74FB-449B-8CA4-1DDF1BA9AA47}.

It occurs to me that when  {177C0AFE-900B-48D4-9E4C-57ADD250B3D4} is installed on 24H2, there will then be TWO identically named decoders with the same DLL file (actually the older 10.00.22621.4391 file will have overwritten the newer 10.00.26100.2161 file),  but with two different CLSIDs.

Is it OK to have two decoders with identical names like this?

**Reply by SimonMacer** [reply]
It's OK. Both CLSIDs are used for the Registry of the AC-3. I only used {177C0AFE-900B-48D4-9E4C-57ADD250B3D4}, since this is the AC-3 MFT from 23H2.

**Comment by arturstat** [comment]
@SimonMacer 
https://drive.google.com/file/d/1kJ9VdAPbAFHi5OSal3stQDWqCIHA6B0J/view?usp=sharing
is not available anymore.
Can you share the reg files again? Please.
Thank you.

**Reply by CeriseRouge0** [reply]
Did you find the reg files or another way to put the data in the registry without thoses files?

**Comment by arturstat** [comment]
@CeriseRouge0 
I did not find any reg files.
I do not know any way to make it work.
I am still waiting for a reply.

**Reply by CeriseRouge0** [reply]
Okay... I've used the standalone exe but I has a bad score on VirusTotal so I want to do it without... but I didn't find any reg files too so it's impossible...

**Reply by SimonMacer** [reply]
> @CeriseRouge0 I did not find any reg files. I do not know any way to make it work. I am still waiting for a reply.

Now, The Microsoft native AC-3 codec (Dolby Digital) and WordPad Feature On Demand (FOD) Package is out. You can follow the guide to re-enable these Feature On Demand (FoD).
https://github.com/SimonMacer/AnWave/releases/tag/Feature-On-Demand

**Comment by MeowConUwU** [comment]
Hi @SimonMacer, dumb question here, but the FoD package that AnWave installed is including both E-AC-3 and AC-3 and decoding and encoding right? Because I only see you mention AC-3 but not E-AC-3.
Thank you!
